crates/async-std-resolver/src/runtime.rs

```diff
@@ -57,31 +57,32 @@ impl Executor for AsyncStdRuntime {
     }
 }
 
-#[derive(Clone, Copy)]
-pub struct AsyncStdRuntimeHandle;
-impl Spawn for AsyncStdRuntimeHandle {
-    fn spawn_bg<F>(&mut self, future: F)
+#[async_trait::async_trait]
+impl RuntimeProvider for AsyncStdRuntime {
+    type UdpSocket = AsyncStdUdpSocket;
+    type Time = AsyncStdTime;
+    type TcpConnection = AsyncStdTcpStream;
+
+    async fn bind_udp(&self, addr: std::net::SocketAddr) -> std::io::Result<Self::UdpSocket> {
+        async_std::net::UdpSocket::bind(addr)
+            .await
+            .map(AsyncStdUdpSocket)
+    }
+
+    async fn connect_tcp(self, addr: std::net::SocketAddr) -> std::io::Result<Self::TcpConnection> {
```

Comment:
Make this `&self` like the other methods to be consistent.

Reply:
Since the returned future's lifetime needs to be disjoint from `self` in this case, the function signature has become a bit messy, but it is now consistent. I'd have preferred to go with an associated type for the future returned by `connect_tcp`, but both tokio and async-std don't have a future type that is used to create a socket, it's just an asynchronous method, so it has to be boxed one way or another.

Reply:
Yeah, this is why I was wondering if there’s value in having this be an `async_trait`, since it might conflict with some of these details.

but yeah, today we can’t avoid Boxed traits, maybe impl traits will eventually work to make the interfaces cleaner (if they don’t now).